### PR TITLE
Keptn Smart Auth: Autofetch keptn endpoint and api-token

### PR DESCRIPTION
**Pull request by ankitjain28may**
Signed-off-by: ankitjain28may <ankitjain28may77@gmail.com>

### CODE REVIEW

File: cli/cmd/auth.go [hunk 1]
```diff
@@ -172,3 +214,32 @@ func authenticate(endPoint string, apiToken string) error {
...
 	return rootCmd.Execute()
 }
+
+func smartKeptnCLIAuth() (string, error) {
+	keptnInstallations, err := keptnutils.GetKeptnManagedNamespace(false)
+	if err != nil {
+		return "", err
+	}
+
+	if len(keptnInstallations) > 1 {
```
Comment (bacherfl) [inline cli/cmd/auth.go:224]: Can we add a parameter to the command to allow to specify one of the available Keptn installations? When used within a script (e.g. an integration test for this feature), the current approach would require a user input during the execution, and cause the script to stall at this point

**Comment by johannes-b** [comment]
- [ ] Can you please add the message "CLI is not authenticated" in case of ClusterIP / NotePort. (to inform the user about the current status)

File: cli/cmd/auth.go (hunk 1 above)
Comment (ankitjain28may) [reply cli/cmd/auth.go:227]: @bacherfl This is done, added a new flag `--ignore-namespace-listing`

Comment (johannes-b) [reply cli/cmd/auth.go:227]: @ankitjain28may great for adding this flag. However, can I ask you a favor to rename it to `--skip-namespace-listing`. The reason is that another command has a similar flag we call `--skip...` (https://github.com/keptn/keptn/issues/2689)  and I would like to have both flags named the same way.

Comment (ankitjain28may) [reply cli/cmd/auth.go:227]: @johannes-b Have made the changes as suggested.

**Review by bacherfl — APPROVED** [review]
lgtm

**Review by bacherfl — CHANGES_REQUESTED** [review]
just saw that one of the Unit tests is failing. Can you take a look at that? Apart from that, the PR looks good from my point of view

**Comment by ankitjain28may** [comment]
@bacherfl The unit tests are failing because it needs this PR to be merged - https://github.com/keptn/kubernetes-utils/pull/13

**Comment by bacherfl** [comment]
> @bacherfl The unit tests are failing because it needs this PR to be merged - [keptn/kubernetes-utils#13](https://github.com/keptn/kubernetes-utils/pull/13)

Ok, got it. Just had one comment about the other PR - once that has been adressed we can merge that one

**Comment by bacherfl** [comment]
> > @bacherfl The unit tests are failing because it needs this PR to be merged - [keptn/kubernetes-utils#13](https://github.com/keptn/kubernetes-utils/pull/13)
> 
> Ok, got it. Just had one comment about the other PR - once that has been adressed we can merge that one

@ankitjain28may I have just merged the kubernetes-utils PR. Can you please update the kubernetes-utils dependency to the latest status from the master branch? When that is done we can merge this PR as well

**Comment by ankitjain28may** [comment]
@bacherfl Have updated the dependency to latest master, Please review :)

**Review by bacherfl — APPROVED** [review]
lgtm

**Review by johannes-b — APPROVED** [review]
lgtm

**Pull request merged**